gramps/plugins/db/dbapi/dbapi.py

```diff
@@ -575,13 +575,15 @@ def set_name_group_mapping(self, name, grouping):
         self.dbapi.execute("SELECT 1 FROM name_group WHERE name = ?",
                            [name])
         row = self.dbapi.fetchone()
+        self._txn_begin()
```

Comment:
Perhaps move this line to the start of the method.

Reply:
Done...